File: src/std/eql.c
#ifndef EQL_H
#define EQL_H

#include "Slice.h"
#include "defs.h"
#include <stdbool.h>

#ifdef BUILDING_WITH_ZIG
extern bool eql(Slice(char) a, Slice(char) b);
#else
static bool eql(Slice(char) a, Slice(char) b) {
  if (a.len != b.len)
    return false;
  if (a.ptr == b.ptr)
    return true;
  for (size_t i = 0; i < a.len; i++) {
    if (a.ptr[i] != b.ptr[i])
      return false;
  }
  return true;
}
#endif

#endif /* EQL_H */

File: src/std/eql.zig
const std = @import("std");

const Slice = @import("Slice.zig").Slice;

pub export fn eql(a: Slice(u8), b: Slice(u8)) bool {
    return std.mem.eql(u8, a.ptr[0..a.len], b.ptr[0..b.len]);
}
